src/glossary.md

```diff
@@ -1,5 +1,34 @@
 # Glossary
 
+## Terms used in the Specification
+
+### Ill-formed program
+
+r[requirement.error]
+
+A Rust program is ill-formed if it violates a static constraint of the language.
+Unless otherwise stated (See ["no diagnostic required"][requirement.ndr]), it is an error to have an ill-formed program.
```

Comment:
> it is an error to have an ill-formed program.

Should that be

> compiling an ill-formed program results in an error diagnostic.

or something like that?

"it is an error" sounds like something the user isn't supposed to do, not necessarily like something that results in an error message.